exporter/exporterhelper/metricshelper.go

```diff
@@ -65,6 +65,9 @@ func NewMetricsExporter(exporterName string, pushMetricsData PushMetricsData, op
 	}
 
 	opts := newExporterOptions(options...)
+	if opts.recordMetrics {
+		pushMetricsData = pushMetricsDataWithMetrics(pushMetricsData)
```

Comment:
Can you please clarify what this is for? It seems like it is recording a metric when exporting a metric fails, but I am not sure and do not quite clearly understand it from the code.

Reply:
This is just a decorator function that we use to avoid duplicating the same logic everywhere. We do have a similar function for trace as well if you look at the entire file.

Reply:
It records two metrics received/dropped timeseries (for metrics) spans (for trace)

Reply:
Understood.